aws/resource_aws_ec2_traffic_mirror_filter_rule.go

```diff
@@ -0,0 +1,356 @@
+				ForceNew: true,
+			},
+			"destination_cidr_block": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateCIDRNetworkAddress,
+			},
+			"destination_port_range": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"from_port": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"to_port": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"protocol": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"rule_action": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					ec2.TrafficMirrorRuleActionAccept,
+					ec2.TrafficMirrorRuleActionReject,
+				}, false),
+			},
+			"rule_number": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"source_cidr_block": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateCIDRNetworkAddress,
+			},
+			"source_port_range": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"from_port": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"to_port": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"traffic_direction": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					ec2.TrafficDirectionIngress,
+					ec2.TrafficDirectionEgress,
+				}, false),
+			},
+		},
+	}
+}
+
+func resourceAwsEc2TrafficMirrorFilterRuleCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	filterId := d.Get("traffic_mirror_filter_id")
+
+	input := &ec2.CreateTrafficMirrorFilterRuleInput{
+		TrafficMirrorFilterId: aws.String(filterId.(string)),
+		DestinationCidrBlock:  aws.String(d.Get("destination_cidr_block").(string)),
+		SourceCidrBlock:       aws.String(d.Get("source_cidr_block").(string)),
+		RuleAction:            aws.String(d.Get("rule_action").(string)),
+		RuleNumber:            aws.Int64(int64(d.Get("rule_number").(int))),
+		TrafficDirection:      aws.String(d.Get("traffic_direction").(string)),
+	}
+
+	if v, ok := d.GetOk("description"); ok {
+		input.Description = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("protocol"); ok {
+		input.Protocol = aws.Int64(int64(v.(int)))
+	}
+
+	if v, ok := d.GetOk("destination_port_range"); ok {
+		input.DestinationPortRange = buildTrafficMirrorPortRangeRequest(v.([]interface{}))
+	}
+
+	if v, ok := d.GetOk("source_port_range"); ok {
+		input.SourcePortRange = buildTrafficMirrorPortRangeRequest(v.([]interface{}))
+	}
+
+	out, err := conn.CreateTrafficMirrorFilterRule(input)
+	if err != nil {
+		return fmt.Errorf("Error creating traffic mirror filter rule for %v", filterId)
```

Comment:
This is not returning the API/SDK error for operators and code maintainers to troubleshoot the issue:

```suggestion
		return fmt.Errorf("error creating EC2 Traffic Mirror Filter Rule (%s): %w", filterId, err)
```